src/mapbox/extended_geolocate_control.js

```diff
@@ -27,6 +28,14 @@ export default class ExtendedGeolocateControl extends GeolocateControl {
   }
 
   async trigger() {
+    const state = await Geolocation.getGeolocationPermission();
+    if (state === Geolocation.geolocationPermissions.PROMPT &&
+        window.localStorage.getItem('hasGeolocateModalOpenedOnce') === null
```

Comment:
is this key explicit enough ?

Reply:
That's probably ok. But it should be namespaced, to avoid confusion or conflict with other Qwant products.  
All interactions with localStorage are currently implemented in "store.js", and I think this logic should be moved there, where the prefix `qmaps_v1` is used in all keys.

Reply:
You r right @amatissart , I refactored store.js a bit to help us reusing the get/set/del functions, and update this pr to use the store. Thanks